### DISCUSSION TITLE
Certificate Issue?

### DISCUSSION THREAD

**Discussion by ClaudiuBadau** [supabase/supabase#3381] [Questions] [answered]
I'm looking to call an endpoint when an insert happens and I'm getting this error
```
**server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none**
```

My function is being called by a trigger on **insert**

```
begin
  perform extensions.http_get('https://webhook.site/f0726ed2-a438-4536-a539-1f7a23b809ba');
end;
```

Any idea why? I was following this discussion -> https://github.com/supabase/supabase/discussions/259

**Comment by dshukertjr** [comment]
Hi @ClaudiuBadau!

I am not certain at all why you are getting that error, but if you are trying to call an endpoint upon database changes, Supabase has a new feature called [function hooks](https://supabase.io/blog/2021/07/30/supabase-functions-updates#function-hooks-alpha) that allows just that! I am not sure if it will solve the problem you are facing, but I hope it helps!

**Reply by Vinzent03** [reply]
I have exact the same issue since today. Same code worked yesterday.
```sql
select content from http(('POST','https://XXX.supabase.co/storage/v1/object/list/objects-data', ARRAY[http_header('authorization','Bearer XXX')],'application/json','{"prefix":"0e8bfe57-cc3e-4919-9c10-a7f0ec2550b2","limit":10,"offset":0,"sortBy":{"column":"name","order":"asc"}}')::http_request);
```
Supabase hooks is not enough, because I need `DELETE`too.

**Reply by ClaudiuBadau** [reply]
Hmm @dshukertjr , that would work but on this project I don't have that option since the project isn't new. I will use that in the future when it's enabled for all projects.

To @Vinzent03 point it, this was working for me too so just weird on why it stopped working.

**Reply by bnjmnt4n** [reply]
Possibly related to Let’s Encrypt’s root certificate expiring: https://scotthelme.co.uk/lets-encrypt-old-root-expiration/. I'm not very familiar with this area, but possibly the boxes that Supabase are hosted on need to update the list of CA certs?

**Reply by inian** [reply]
Yup the Lets encrypt root expiring might be the issue. Can you send the project reference to support@supabase.io? Also which device are you making the request from?

**Reply by mike-wax** [reply]
@inian Having this same issue (please see below). What do you need to resolve it?

**Comment by mike-wax** [comment]
Looks like this started happening around 2021-09-30T14:02:01.013Z

I have a pg_cron that started failing then for this error



```
| jobid | runid | job_pid | database | username       | command                                                                                                                                                                                                                | status | return_message                                                              | start_time               | end_time                 |
| ----- | ----- | ------- | -------- | -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ | --------------------------------------------------------------------------- | ------------------------ | ------------------------ |
| 1     | 19682 | 323389  | postgres | supabase_admin | 
    select content
    from
      http_post(
        'https://wax.run/api', -- webhook URL, replace the ID(223c8..) with your own
        '{"cronTwitter": true}', -- payload
        'application/json'
      )
     | failed | ERROR:  server certificate verification failed. CAfile: none CRLfile: none
 | 2021-09-30T14:02:01.013Z | 2021-09-30T14:02:01.051Z |
```

**Comment by inian** [comment, marked as answer]
This should due to Lets Encrypt root cert expiring and should be solved now. If you are still facing this issue, reach out.

**Reply by Zara603** [reply]
I am having this issue today and was working yesterday, my email is zlotusa@gmail.com

**Reply by vishaltarkar** [reply]
I'm getting this error

`Error: The certificate used to secure the TLS connection is invalid.
    at Connection.#startup (https://deno.land/x/postgres@v0.17.0/connection/connection.ts:392:19)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async Connection.startup (https://deno.land/x/postgres@v0.17.0/connection/connection.ts:491:11)
    at async PoolClient.connect (https://deno.land/x/postgres@v0.17.0/client.ts:220:7)
    at async DeferredAccessStack.pop (https://deno.land/x/postgres@v0.17.0/utils/deferred.ts:121:7)
    at async file:///src/main.ts:13:24
    at async ext:deno_http/00_serve.js:455:18`
    
 Error just started coming from yesterday, was working fine.
 
 Changes, I have made, upgrade my server from micro to small and changed my db password.

```
const databaseUrl = Deno.env.get('SUPABASE_DB_URL');

const pool = new postgres.Pool(databaseUrl, 3, true)

 const connection = await pool.connect()

```


Any help or guidance will be very much appriciated.

Thank you!

**Reply by ptts** [reply]
Hey @inian, almost exactly 4 years later and the problem still exists 🥲
https://github.com/orgs/supabase/discussions/12401#discussioncomment-14616139

We contacted your support 7 days ago, but the certificate is still expired.

Anything we can do manually to trigger a new certificate generation for our DB?

**Comment by Shaun-Regenbaum** [comment]
I am also having this issue.

**Comment by tmountain** [comment]
Having this issue now.

**Reply by dshukertjr** [reply]
Could you contact supabase.help for this?

**Comment by hitsthings** [comment]
For future travelers: https://github.com/orgs/supabase/discussions/12401#discussioncomment-12443835

SSL certificates expire 2 years after your project was created and support does not respond for 48 hours even if you're on a Pro plan. Contacting support is not the answer for a systemic issue with SSL certificates.
